src/device/rttDevice.js

```diff
@@ -441,7 +439,10 @@ class RTTDevice extends Device {
             const value =
                 this.getAdcResult[currentMeasurementRange](adcResult) * 1e6;
 
-            this.onSampleCallback({ value, timestamp: this.timestamp });
+            this.onSampleCallback({
+                value,
+                endOfTrigger: i === this.dataPayload.length - 2,
```

Comment:
I don't really understand this tbh, what's going on here? Otherwise it looks great!

Reply:
Well, since the loop goes from 0 to _this.dataPayload.length - 1_ by steps of 2, _i_ will be _this.dataPayload.length - 2_ at the last iteration.